Tests/Services/MockServiceTests.swift

```diff
@@ -126,12 +156,12 @@ class MockServiceTests: XCTestCase {
 				if let data = data {
 					let testResult = self.decodeResponse(data: data)
 					XCTAssertEqual(testResult?.foo, "bar")
+					completion()
+					testExpectation.fulfill()
 				}
 				if let error = error {
 					XCTFail(error.localizedDescription)
 				}
-				completion()
-				testExpectation.fulfill()
```

Comment:
Now sure why you changed this, the expectation is to receive a response, if the value of the response is bad, than the assertion should fail.
With this change, in case you get an error, the entire test will stall until the timeout will pass, and it will also fail from the `XCTFail`.
I see you did this change for other tests as well.

Reply:
Found out there were false positives when expecting a certain status code but endpoint was not hit.